libs/velo-external-db-core/src/service/schema.spec.ts

```diff
@@ -71,20 +72,235 @@ describe('Schema Service', () => {
+            expect(compareColumnsInDbAndRequest(columnsInDb, [...columnsInRequest, newColumn]).columnsToAdd).toEqual(convertWixFormatFieldsToInputFields([newColumn]))
+        })
+
+        test('compareColumnsInDbAndRequest function - remove columns', async() => {
+            const { compareColumnsInDbAndRequest } = env.schemaService
+            const columnsInDb = [{
+                field: ctx.column.name,
+                type: ctx.column.type
+            }]
+            const columnsInRequest = [{
+                key: ctx.column.name,
+                type: convertFieldTypeToEnum(ctx.column.type),
+            }]
+            const newColumn = {
+                key: ctx.anotherColumn.name,
+                type: convertFieldTypeToEnum(ctx.anotherColumn.type)
+            }
+            expect(compareColumnsInDbAndRequest([], []).columnsToRemove).toEqual([])
+            expect(compareColumnsInDbAndRequest(columnsInDb, columnsInRequest).columnsToRemove).toEqual([])
+            expect(compareColumnsInDbAndRequest(columnsInDb, [...columnsInRequest, newColumn]).columnsToRemove).toEqual([])
+            expect(compareColumnsInDbAndRequest(columnsInDb, []).columnsToRemove).toEqual(columnsInDb.map(f => f.field))
+            expect(compareColumnsInDbAndRequest(columnsInDb, [newColumn]).columnsToRemove).toEqual(columnsInDb.map(f => f.field))
+        })
+
+        test('compareColumnsInDbAndRequest function - change column type', async() => {
+            const { compareColumnsInDbAndRequest } = env.schemaService
+            const columnsInDb = [{
+                field: ctx.column.name,
+                type: 'text'
+            }]
+
+            const columnsInRequest = [{
+                key: ctx.column.name,
+                type: convertFieldTypeToEnum('text'),
+            }]
+
+            const changedColumnType = {
+                key: ctx.column.name,
+                type: convertFieldTypeToEnum('number')
+            }
+
+            expect(compareColumnsInDbAndRequest([], []).columnsToChangeType).toEqual([])
+            expect(compareColumnsInDbAndRequest(columnsInDb, columnsInRequest).columnsToChangeType).toEqual([])
+            expect(compareColumnsInDbAndRequest(columnsInDb, [changedColumnType]).columnsToChangeType).toEqual(convertWixFormatFieldsToInputFields([changedColumnType]))
+
+        })
+
+        test('update collection - add new columns', async() => { 
+            const newFields = [{
+                key: ctx.column.name,
+                type: convertFieldTypeToEnum(ctx.column.type),
+            }]
+
+            driver.givenAllSchemaOperations()
+            schema.expectSchemaRefresh()            
+            driver.givenFindResults([ { id: ctx.collectionName, fields: [] } ])
+
+            await env.schemaService.updateCollection({ id: ctx.collectionName, fields: newFields })
+
+            expect(driver.schemaProvider.addColumn).toBeCalledTimes(1)    
```

Comment:
not enough to check that have been called, need to check that added the right column.
`toBeCalledWith(collectionName, SpiFieldToDomainField(ctx.field)`